pkg/unittest/validators/common.go

```diff
@@ -23,24 +23,23 @@ type SnapshotComparer interface {
 
 // ValidateContext the context passed to validators
 type ValidateContext struct {
-	Docs     []common.K8sManifest
-	Index    int
-	Negative bool
+	Docs         []common.K8sManifest
+	SelectedDocs *[]common.K8sManifest
```

Comment:
The core of the solution is that instead of relying on the "Index" of selected document we simply pass the array of "selected" documents to the assertion context. 

I believe it's a bit simpler design to operate on the documents array directly instead of dealing with Indices array which then would have to be validated against  indices going outside of boundary.

The context also includes unfiltered list of all documents (`Docs`) which makes assertions such as `hasDocuments` to stil work fine ignoring the by ignoring the Selection.